packages/medusa/src/api/routes/admin/products/transaction/create-product-variant.ts

```diff
@@ -78,51 +78,78 @@ export const createVariantTransaction = async (
 
   const productVariantServiceTx = productVariantService.withTransaction(manager)
 
-  async function createVariant(variantInput: CreateProductVariantInput) {
+  async function createVariants(variantInput: CreateProductVariantInput[]) {
     return await productVariantServiceTx.create(productId, variantInput)
   }
 
-  async function removeVariant(variant: ProductVariant) {
-    if (variant) {
-      await productVariantServiceTx.delete(variant.id)
+  async function removeVariants(variants: ProductVariant[]) {
+    if (variants.length) {
+      await productVariantServiceTx.delete(variants.map((v) => v.id))
     }
   }
 
-  async function createInventoryItem(variant: ProductVariant) {
-    if (!variant.manage_inventory) {
-      return
-    }
-
-    return await inventoryService!.createInventoryItem({
-      sku: variant.sku,
-      origin_country: variant.origin_country,
-      hs_code: variant.hs_code,
-      mid_code: variant.mid_code,
-      material: variant.material,
-      weight: variant.weight,
-      length: variant.length,
-      height: variant.height,
-      width: variant.width,
-    })
+  async function createInventoryItems(variants: ProductVariant[] = []) {
+    const context = { transactionManager: manager }
+
+    return await Promise.all(
+      variants.map(async (variant) => {
+        if (!variant.manage_inventory) {
+          return
+        }
+
+        const inventoryItem = await inventoryService!.createInventoryItem(
```

Comment:
question: Are we keeping the refactoring of the inventory service (to support passing bulk input) for a separate PR?

Reply:
Yea, i ve moved everything in that flow to take bulk and centralised the last piece which is the inventory. Though, i would like to have it in a separate pr as it is a separate domain and have a time to merge quicker

Reply:
Sounds reasonable to me 👍